cmd/distro.go

```diff
@@ -678,3 +685,72 @@ func NewDistroReportCmd() (*cobra.Command, error) {
 	distroReportCmd.Flags().String("name", "", "the distro name")
 	return distroReportCmd, nil
 }
+
+func NewDistroExportCmd() (*cobra.Command, error) {
+	distroExportCmd := &cobra.Command{
+		Use:   "export",
+		Short: "export distributions",
+		Long:  `Export distributions.`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			formatOption, err := cmd.Flags().GetString("format")
+			if err != nil {
+				return err
+			}
+			if formatOption != "json" && formatOption != "yaml" {
+				return fmt.Errorf("format must be json or yaml")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := generateCobblerClient()
+			if err != nil {
+				return err
+			}
+
+			name, err := cmd.Flags().GetString("name")
+			if err != nil {
+				return err
+			}
+			formatOption, err := cmd.Flags().GetString("format")
+			if err != nil {
+				return err
+			}
+
+			itemNames := make([]string, 0)
+			if name == "" {
+				itemNames, err = Client.ListDistroNames()
+				if err != nil {
+					return err
+				}
+			} else {
+				itemNames = append(itemNames, name)
+			}
+
+			for _, itemName := range itemNames {
+				distro, err := Client.GetDistro(itemName, false, false)
+				if err != nil {
+					return err
+				}
+				if formatOption == "json" {
+					jsonDocument, err := json.Marshal(distro)
+					if err != nil {
+						return err
+					}
+					fmt.Fprintln(cmd.OutOrStdout(), string(jsonDocument))
+				}
+				if formatOption == "yaml" {
+					yamlDocument, err := yaml.Marshal(distro)
+					if err != nil {
+						return err
+					}
+					fmt.Fprintln(cmd.OutOrStdout(), "---")
```

Comment:
Small nitpick: I assume this is to separate the JSON from the YAML output. Is it even possible to get both at once? And I would prefer to keep it consistent between JSON and YAML

Reply:
This is to seperate the different YAML documents in case multiple systems are exported.

Reply:
I still think this could be done a bit smarter. Like joining all of the objects into one string with this line as a separator and printing that, but I won't insist that you change it if you think this solution is fine

Reply:
This is the marker that is used in the official YAML spec: https://yaml.org/spec/1.2.2/#912-document-markers